eventuals/task.h

```diff
@@ -570,57 +598,66 @@ class TaskFromToWith {
 
 ////////////////////////////////////////////////////////////////////////
 
-struct Task final {
-  template <typename From_>
-  struct From final : public TaskFromToWith<From_, void> {
-    template <typename To_>
-    struct To final : public TaskFromToWith<From_, To_> {
-      template <typename... Args_>
-      using With = TaskFromToWith<From_, To_, Args_...>;
-
-      To(MonostateIfVoidOr<To_> value)
-        : TaskFromToWith<From_, To_>(std::move(value)) {}
-
-      template <typename F>
-      To(F f)
-        : TaskFromToWith<From_, To_>(std::move(f)) {}
+template <
+    typename From_ = void,
+    typename To_ = void,
+    typename Errors_ = std::tuple<>,
+    typename... Args_>
+struct TaskBuilder final
+  : public TaskFromToWith<From_, To_, Errors_, Args_...> {
+  template <typename T>
+  using From = TaskBuilder<T, To_, Errors_, Args_...>;
 
-      To(To&& that) = default;
+  template <typename T>
+  using To = TaskBuilder<From_, T, Errors_, Args_...>;
 
-      ~To() override = default;
-    };
+  template <typename T>
+  using Of = TaskBuilder<void, T, Errors_, Args_...>;
```

Comment:
Since `Of` is an alias for both `From` and `To` then we can check in the `std::enable_if_t` as appropriate!